packages/protocol/contracts/L1/libs/LibTokenomics.sol

```diff
@@ -53,107 +56,100 @@ library LibTokenomics {
+        return state.basefee;
     }
 
     function getProofReward(
         TaikoData.State storage state,
         TaikoData.Config memory config,
         uint64 provenAt,
         uint64 proposedAt
+    ) internal view returns (uint64 reward) {
+        (reward, , ) = calculateBasefee(state, config, (provenAt - proposedAt));
+    }
+
+    /**
+     * Update the baseFee for proofs
+     *
+     * @param state The actual state data
+     * @param config Config data
+     * @param proofTime The actual proof time
+     * @return reward Amount of reward given - if blocked is proved and verified
+     * @return newProofTimeIssued Accumulated proof time
+     * @return newBasefee New basefee
+     */
+    function calculateBasefee(
+        TaikoData.State storage state,
+        TaikoData.Config memory config,
+        uint64 proofTime
     )
         internal
         view
-        returns (uint64 newFeeBase, uint64 reward, uint64 premiumRate)
+        returns (uint64 reward, uint64 newProofTimeIssued, uint64 newBasefee)
     {
-        if (proposedAt > provenAt) revert L1_INVALID_PARAM();
-
-        (newFeeBase, premiumRate) = getTimeAdjustedFee({
-            feeConfig: config.provingConfig,
-            feeBase: state.feeBase,
-            isProposal: false,
-            timeUsed: provenAt - proposedAt,
-            timeAverage: state.avgProofTime
-        });
-        reward = getSlotsAdjustedFee({
-            state: state,
-            config: config,
-            isProposal: false,
-            feeBase: newFeeBase
-        });
+        newProofTimeIssued = state.proofTimeIssued;
 
-        unchecked {
-            reward = uint64((reward * (10000 - config.rewardBurnBips)) / 10000);
-        }
-    }
+        newProofTimeIssued = (newProofTimeIssued > config.proofTimeTarget)
+            ? newProofTimeIssued - config.proofTimeTarget
+            : uint64(0);
+        newProofTimeIssued += proofTime;
 
-    // Implement "Slot-availability Multipliers", see the whitepaper.
-    function getSlotsAdjustedFee(
-        TaikoData.State storage state,
-        TaikoData.Config memory config,
-        bool isProposal,
-        uint64 feeBase
-    ) internal view returns (uint64) {
-        unchecked {
-            // m is the `n'` in the whitepaper
-            uint256 m = 1000 *
-                config.maxNumProposedBlocks +
-                config.slotSmoothingFactor;
-            // n is the number of unverified blocks
-            uint256 n = 1000 *
-                (state.numBlocks - state.lastVerifiedBlockId - 1);
-            // k is `m − n + 1` or `m − n - 1`in the whitepaper
-            uint256 k = isProposal ? m - n - 1000 : m - n + 1000;
-            return uint64((feeBase * (m - 1000) * m) / (m - n) / k);
+        newBasefee = _calcBasefee(
+            newProofTimeIssued,
+            config.proofTimeTarget,
+            config.adjustmentQuotient
+        );
+
+        uint64 numBlocksBeingProven = state.numBlocks -
```

Comment:
actually I dont think this value (`state.numBlocks - state.lastVerifiedBlockId - 1`) can represent `numBlocksBeingProven`, its just `numBlocksUnverified`, lets say there are 100 unproven blocks (height 101 to height 200), then we submit the proofs for blocks from height 102 to 200, and then this value will be still 100 (since block 101 has not been proven yet).

Reply:
Fair point !

Reply:
Hmm, the more I think about it the more I tend to think does not change anything.

It is indeed not the `numBlocksBeingProven`, but we could call it like `numBlocksUnverified` or even `numBlocksUnpaidRewards` which is a valid representation what we would like to achieve. Every update in basefee and reward is done during `verifyBlock`, as well as the payouts.
@Brechtpd  ?